spec/compiler/parser/parser_spec.cr

```diff
@@ -1849,6 +1849,14 @@ module Crystal
 
       it_parses %(annotation Foo\nend\nrequire "bar"), [AnnotationDef.new("Foo".path), Require.new("bar")]
 
+      it_parses "def foo(x,y);end;foo = 10;puts foo -a, b", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", [Call.new("foo".var, "-", "b".var), "b".var] of ASTNode)]
+      it_parses "def foo(x,y);end;foo = 10;puts(foo -a, b)", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", [Call.new("foo".var, "-", "b".var), "b".var] of ASTNode)]
+      it_parses "def foo(x,y);end;foo = 10;puts(foo +a, b)", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", [Call.new("foo".var, "+", "b".var), "b".var] of ASTNode)]
+      it_parses "def foo(x,y);end;foo = 10; puts foo a, b", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", Call.new(nil, "foo", ["a".var, "b".var] of ASTNode))]
+      it_parses "def foo(x,y);end;foo = 10; puts(foo a, b)", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", Call.new(nil, "foo", ["a".var, "b".var] of ASTNode))]
+      it_parses "def foo(x,y);end;foo = 10; puts foo -a, b: 2", [Def.new("foo", ["x".arg, "y".arg] of Arg), Assign.new("foo".var, 10.int32), Call.new(nil, "puts", [Call.new("foo".var, "-", "a".var), NamedArgument.new("b", 2.int32)] of ASTNode)]
+      it_parses "def foo(x,y);end; puts foo -a, b", [Def.new("foo", ["x".arg, "y".arg] of Arg), Call.new(nil, "puts", Call.new(nil, "foo", ["a".var, "b".var] of ASTNode))]
```

Comment:
There's no need to define the `def`s here, they don't affect the parsing.

Reply:
I know, I just kept following the suggestion I got in #9194. But I can remove it if you want

Reply:
Oh, but that suggestion doesn't make sense because the parser doesn't care about methods defined. It would be pretty catastrophic if parsing depended on this. So there's no need at all do it.